src/shell/affinity.c

```diff
@@ -41,6 +66,89 @@ static int topology_restrict (hwloc_topology_t topo, hwloc_cpuset_t set)
     return (0);
 }
 
+
+/*  Parse a list of hwloc bitmap strings in list, bitmask, or taskset
+ *  form and return an allocated hwloc_cpuset_t array of size ntasks,
+ *  filled with the resulting bitmasks. If ntasks is greater than the number
+ *  of provided cpusets, then cpusets are reused as necessary.
+ *
+ *  It is an error if any cpuset does not fall within job_cpuset.
+ */
+static hwloc_cpuset_t *parse_cpuset_list (const char *setlist,
+                                          hwloc_cpuset_t job_cpuset,
+                                          int ntasks)
+{
+    char *copy = NULL;
+    char *s, *arg, *sptr = NULL;
+    int index, i = 0;
+    hwloc_cpuset_t *cpusets = NULL;
+
+    if (!(cpusets = cpuset_array_create (ntasks))
+        || !(copy = strdup (setlist))) {
+        shell_log_errno ("out of memory");
+        goto err;
+    }
+
+    s = copy;
+    while ((arg = strtok_r (s, ";", &sptr)) && i < ntasks) {
+        int rc;
+        if (strstarts (arg, "0x")) {
+            /*  If string starts with 0x then parse as a bitmask. If a
+             *  comma is present in the string, then this is likely a
+             *  hwloc-style bitmap string, otherwise, try the taskset
+             *  style bitmaps, which are simpler.
+             */
+            if (strchr (arg, ','))
+                rc = hwloc_bitmap_sscanf (cpusets[i], arg);
+            else
+               rc = hwloc_bitmap_taskset_sscanf (cpusets[i], arg);
```

Comment:
indent problem there?

Reply:
Nice catch! Fixed, rebased, and pushed. I'll set MWP.